extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/devservices/DevServicesJaegerProcessor.java

```diff
@@ -0,0 +1,240 @@
+            log.warn("Please configure '" + OTEL_CONFIG_ENDPOINT + "' or get a working docker instance");
+            return null;
+        }
+
+        DockerImageName dockerImageName = DockerImageName
+                .parse(capturedDevServicesConfiguration.imageName().orElse(JAEGER_IMAGE))
+                .asCompatibleSubstituteFor(JAEGER_IMAGE);
+
+        final Supplier<RunningDevService> defaultJaegerSupplier = () -> {
+            QuarkusJaegerContainer jaegerContainer = new QuarkusJaegerContainer(dockerImageName,
+                    capturedDevServicesConfiguration.port(),
+                    capturedDevServicesConfiguration.serviceName(), useSharedNetwork);
+            timeout.ifPresent(jaegerContainer::withStartupTimeout);
+            jaegerContainer.start();
+
+            String jaegerHost = JAEGER_SCHEME + jaegerContainer.getHost() + ":" + jaegerContainer.getPort() + JAEGER_ENDPOINT;
+
+            return new RunningDevService(Feature.OPENTELEMETRY_JAEGER_EXPORTER.getName(), jaegerContainer.getContainerId(),
+                    new ContainerShutdownCloseable(jaegerContainer, "Jaeger"),
+                    OTEL_CONFIG_ENDPOINT, jaegerHost);
+        };
+
+        return jaegerContainerLocator
+                .locateContainer(capturedDevServicesConfiguration.serviceName(), capturedDevServicesConfiguration.shared(),
+                        LaunchMode.current())
+                .map(containerAddress -> {
+                    String jaegerUrl = JAEGER_SCHEME + containerAddress.getUrl() + JAEGER_ENDPOINT;
+                    return new RunningDevService(Feature.OPENTELEMETRY_JAEGER_EXPORTER.getName(), containerAddress.getId(),
+                            null, OTEL_CONFIG_ENDPOINT, jaegerUrl);
+                })
+                .orElseGet(defaultJaegerSupplier);
+    }
+
+    private static class QuarkusJaegerContainer extends GenericContainer<QuarkusJaegerContainer> {
+        private final OptionalInt fixedExposedPort;
+        private final boolean useSharedNetwork;
+
+        private String hostName = null;
+
+        public QuarkusJaegerContainer(DockerImageName dockerImageName, OptionalInt fixedExposedPort, String serviceName,
+                boolean useSharedNetwork) {
+            super(dockerImageName);
+            this.fixedExposedPort = fixedExposedPort;
+            this.useSharedNetwork = useSharedNetwork;
+
+            this.waitStrategy = (new LogMessageWaitStrategy()).withRegEx(
+                    ".*\"Health Check state change\",\"status\":\"ready\".*")
+                    .withStartupTimeout(Duration.of(15L, ChronoUnit.SECONDS));
+
+            if (serviceName != null) {
+                withLabel(DEV_SERVICE_LABEL, serviceName);
+            }
+        }
+
+        @Override
+        public void close() {
+
+        }
+
+        @Override
+        protected void configure() {
+            super.configure();
+
+            addFixedExposedPort(4318, 4318);
```

Comment:
Using these fixed ports means it won't work correctly with remote Docker agents (e.g. I have slow network and use TestContainers Cloud when in hurry). Any reason why don't use simply wait and get mapped port?